pool/web/handler/contact_assignment.ml

```diff
@@ -52,16 +57,22 @@ let create req =
            database_label
            experiment.Experiment.Public.id
            contact
-         ||> CCOption.is_some
+         ||> CCList.is_empty
+         ||> not
        in
-       let tags = Logger.req req in
        let events =
-         Cqrs_command.Assignment_command.Create.(
-           handle
-             { contact; session; waiting_list; experiment }
-             tenant
-             confirmation_email
-             already_enrolled)
+         let open Cqrs_command.Assignment_command.Create in
+         CCList.map
+           (fun (session, waiting_list) ->
+             handle
+               ~tags
+               { contact; session; waiting_list; experiment }
+               tenant
+               confirmation_email
+               already_enrolled)
+           ((session, waiting_list) :: CCList.map (fun m -> m, None) follow_ups)
```

Comment:
Is the waiting list supposed to be `None` all the time?

Reply:
We only need to remove the subject from the waiting list once (within the first iteration, where it isn't `None`).